shell/components/dialog/ConfirmRelatedToRemoveDialog.vue

```diff
@@ -0,0 +1,166 @@
+<script>
+import { mapState } from 'vuex';
+import { exceptionToErrorsArray } from '@shell/utils/error';
+import { alternateLabel } from '@shell/utils/platform';
+import AsyncButton from '@shell/components/AsyncButton';
+import { Banner } from '@components/Banner';
+import { Card } from '@components/Card';
+import { escapeHtml } from '@shell/utils/string';
+import CopyToClipboardText from '@shell/components/CopyToClipboardText';
+
+/**
+ * @name ConfirmRelatedToRemoveDialog
+ * @description Dialog component to confirm the related resources before removing the resource.
+ */
+export default {
+  name: 'ConfirmRelatedToRemoveDialog',
+
+  components: {
+    AsyncButton,
+    Banner,
+    Card,
+    CopyToClipboardText
+  },
+
+  props:  {
+    /**
+     * @property resources to be deleted.
+     * @type {Resource[]} Array of the resource model's instance
+     */
+    resources: {
+      type:     Array,
+      required: true
+    }
+  },
+
+  data() {
+    return { errors: [], confirmName: '' };
+  },
+
+  computed: {
+    ...mapState('action-menu', ['modalData']),
```

Comment:
Seems `action-menu` unused?

Reply:
I think `action-menu` like the module/namespace of `modalData`, you cannot get modalData state if you do not declare the module/namespace